front-end/src/app/shared/interfaces/transaction-type.interface.ts

```diff
@@ -11,9 +13,18 @@ export interface TransactionType {
   isDependentChild: boolean; // When set to true, the parent transaction is used to generate UI form entry page
   title: string;
   schema: JsonSchema; // FEC validation JSON schema
-  transaction: Transaction | undefined;
-  parentTransaction: Transaction | undefined;
-  childTransactionType: TransactionType | undefined;
+  transaction?: Transaction;
+  parentTransaction?: Transaction;
+  childTransactionType?: TransactionType;
+  subTransactionTypes?: ScheduleATransactionTypes[];
```

Comment:
Would you add an inline comment describing why the subTransactoinTypes[] is needed? It's not so obvious without digging through the code.

Reply:
on it!